src/communication.ts

```diff
@@ -0,0 +1,65 @@
+import { io, Socket } from 'socket.io-client';
+import { randomBytes } from './util/crypto';
+
+export class Communication {
+    socketServer: Socket;
+
+    constructor() {
+        const url = process.env.REACT_APP_COMMUNICATION_URL ?? 'ws://ms.tonomy.staging';
+        console.log(`connecting to service ${url}...`);
+        this.socketServer = io(url, {
+            transports: ['websocket'],
+        });
+    }
+
+    /**
+     * connect unregistered user to the website
+     * @param randomSeed the random seed the user need to connect on typically recieved in jwt
+     */
+    connectTonomy(randomSeed: string) {
+        this.socketServer.emit('connectTonomy', { randomSeed });
+    }
+
+    onClientConnected(func: (param?: any) => void) {
```

Comment:
for now, lets have 3 interfaces:
login - logs in
createChannel - creates a channel
sendMessage - sends a message
subscribeMessages - subscribes to all messages for a channel sent by sendMessage

sendMessage and subscribeMessages are agnostic to what is sent at this point
internally in the message we can specify what function is called when the recipient gets it. this is a more flexible approach for the future

Reply:
this is possible once we dont have to connect on channels